coordinator/src/services/translation/line-item.ts

```diff
@@ -57,27 +59,32 @@ export function convertPrescriptionEndorsements(fhirMedicationRequest: fhir.Medi
   }
 }
 
-function convertAdditionalInstructions(fhirMedicationRequest: fhir.MedicationRequest, patientInfoStr: string)  {
+function convertAdditionalInstructions(fhirMedicationRequest: fhir.MedicationRequest, patientInfoText: Array<core.Text>)  {
   const controlledDrugWordsExtension = getExtensionForUrlOrNull(
     fhirMedicationRequest.dispenseRequest.extension,
     "https://fhir.nhs.uk/R4/StructureDefinition/Extension-controlled-drug-quantity-words",
     "MedicationRequest.dispenseRequest.extension"
   ) as fhir.StringExtension
   const controlledDrugWords = controlledDrugWordsExtension?.valueString
-  const controlledDrugStr = controlledDrugWords ? `CD: ${controlledDrugWords}\n` : ""
+  const controlledDrugWordsWithPrefix = controlledDrugWords ? `CD: ${controlledDrugWords}` : ""
 
   const patientInstruction = onlyElement(
     fhirMedicationRequest.dosageInstruction,
     "MedicationRequest.dosageInstruction"
   ).patientInstruction
-  const patientInstructionStr = patientInstruction ? patientInstruction : ""
 
-  const additionalInstructionsValue = `${patientInfoStr}${controlledDrugStr}${patientInstructionStr}`
-  const hl7V3AdditionalInstructions = new prescriptions.AdditionalInstructions(additionalInstructionsValue)
+  const additionalInstructionsValueObj = {} as ElementCompact
+  if (patientInfoText?.length) {
+    additionalInstructionsValueObj.patientInfo = patientInfoText
+  }
+  additionalInstructionsValueObj._text = [controlledDrugWordsWithPrefix, patientInstruction].filter(Boolean).join("\n")
+  const additionalInstructionsValueStr = js2xml(additionalInstructionsValueObj, {compact: true})
```

Comment:
does the js2xml maintain the order of elements as we had in the previous version of translation?

Reply:
and is the ordering important